web3/contract/utils.py

```diff
@@ -129,6 +132,10 @@ def call_contract_function(
     )
     normalized_data = map_abi_data(_normalizers, output_types, output_data)
 
+    if decode_tuples:
```

Comment:
This seems like the ideal behavior for all cases. If requested, convert to `ABIDecodedNamedTuple` types.